packages/compass-web/src/url-builder.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { expect } from 'chai';
+import {
+  getRouteFromWorkspaceTab,
+  getWorkspaceTabFromRoute,
+} from './url-builder';
+
+describe('url builder helpers', function () {
+  describe('getRouteFromWorkspaceTab', function () {
+    const specs = [
+      ['Welcome', '/', null],
+      ['Databases', '/Cluster0', null],
+      ['Collections', '/Cluster0/%23db', { namespace: '#db' }],
+      ['Collection', '/Cluster0/%23db/coll', { namespace: '#db.coll' }],
+      [
+        'Collection',
+        '/Cluster0/%23db/coll/find',
+        { namespace: '#db.coll', initialSubtab: 'Documents' },
+      ],
+      [
+        'Collection',
+        '/Cluster0/%23db/coll/aggregation',
+        { namespace: '#db.coll', initialSubtab: 'Aggregations' },
+      ],
+      [
+        'Collection',
+        '/Cluster0/%23db/coll/schema',
+        { namespace: '#db.coll', initialSubtab: 'Schema' },
+      ],
+      [
+        'Collection',
+        '/Cluster0/%23db/coll/indexes',
+        { namespace: '#db.coll', initialSubtab: 'Indexes' },
+      ],
+      [
+        'Collection',
+        '/Cluster0/%23db/coll/validation',
+        { namespace: '#db.coll', initialSubtab: 'Validation' },
+      ],
+      ['Collection', '/Cluster0/%23db/coll/foobar', { namespace: '#db.coll' }],
+    ] as const;
+
+    for (const [type, route, extraParams] of specs) {
+      it(`should return ${type} workspace when initial route is ${route}`, function () {
+        expect(getWorkspaceTabFromRoute(route)).to.deep.eq({
+          type,
+          ...(type !== 'Welcome' && { connectionId: 'Cluster0' }),
+          ...extraParams,
+        });
+      });
+    }
+  });
+
+  describe('getRouteFromWorkspaceTab', function () {
```

Comment:
is it allowed to have multiple describe blocks with same title 🤔 ?

Reply:
Haha, apparently 🙈 

![image](https://github.com/user-attachments/assets/706075f1-9957-49f6-a6da-93aab61f2a98)

Thanks for spotting, I'll update